R/checks.R

```diff
@@ -24,20 +25,63 @@
 check_user_rights <- function(db_data,
                               db_metadata,
                               call = caller_env()) {
-  missing_db_metadata <- db_metadata %>% # nolint: object_usage_linter
-    filter(!.data$field_name_updated %in% names(db_data)) %>%
+  # Set condition components used in both error and warning
+  msg_info <- c("i" = "This can happen when the user privileges are not set to allow exporting certain
+    instruments via the API.")
+  cond_class <- c("redcap_user_rights", "REDCapTidieR_cond")
+
+  db_fields <- names(db_data)
+  # Fields in metadata excluding record ID
+  metadata_fields <- setdiff(db_metadata$field_name_updated, get_record_id_field(db_data))
+
+  # If no data fields were returned the user has no data access
+  if (length(intersect(db_fields, metadata_fields)) == 0) {
+    cli_abort(
+      c(
+        "x" = "The REDCap API returned no data for any of the requested instruments.",
+        msg_info
+      ),
+      class = c("no_data_access", cond_class),
+      call = call
+    )
+  }
+
+  missing_db_metadata <- db_metadata %>%
     select("field_name_updated", "form_name") %>%
     group_by(.data$form_name) %>%
-    summarise(fields = list(.data$field_name_updated))
+    filter(!all(.data$field_name_updated %in% db_fields)) |>
```

Comment:
Ooo good catch. Fixing this.

I did a find all to make sure and it looks like the only place we have `|>` is in the vignettes. Does that matter? Should we switch to `%>%` there to be safe?

Reply:
It probably doesn't matter for the vignettes, since those will generally show REDCapTidieR in use with other things, but I remember we talked about keeping `%>%` for a while yet so that we don't have to force users to rely on newer versions of R.